### PR TITLE
Adding Node FaceLocator template & made fixes to FaceLocator c#

### PR DESCRIPTION
**Pull request by soninaren**

### CODE REVIEW

**Comment by soninaren** [comment]
fixes #222

File: Templates/FaceLocator-NodeJS/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            response = JSON.parse(response);
+
+            if (typeof response == 'undefined' && typeof response.faces.length === 0) {
+                context.done();
+                return;
+            }
+            context.log("From Vision Api:", response);
+
+            var guid = require('node-uuid');
+            var faces = [];
```
Comment (mamaso) [inline Templates/FaceLocator-NodeJS/index.js]: I'd like a more functional style, for loops are meh.  The object.assign is questionable, but it's shorter.

```js
var faces = response.faces.map((face) => {
  var faceMapping = {
    rowKey: ...,
    imageFile: ...,
    partitionKey: ...
  }
  Object.assign(faceMapping, face.faceRectangle);
  return faceMapping;
});

File: Templates/FaceLocator-NodeJS/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            'Content-Type': 'application/octet-stream',
+            'Ocp-Apim-Subscription-Key': process.env.Vision_API_Subscription_Key
+        }
+    };
+
+    request(options)
+        .then(function (response) {
+            response = JSON.parse(response);
+
+            if (typeof response == 'undefined' && typeof response.faces.length === 0) {
```
Comment (mamaso) [inline Templates/FaceLocator-NodeJS/index.js]: bit shorter

```js
if (!response || !response.faces.length)
  return context.done()
```

File: Templates/FaceLocator-NodeJS/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+// 2) Go to Function app settings -> App Service settings -> Settings -> Application settings
+//    create a new app setting Vision_API_Subscription_Key and use Computer vision key as value
+// 3) Go to Function app settings -> App Service settings -> Tools -> Console
+//    Enter the following commands: 
+//    > cd <functionName>
+//    > npm install
+module.exports = function (context, image) {
+    context.log("Image Size:", image.length);
+
+    var request = require('request-promise');
```
Comment (mamaso) [inline Templates/FaceLocator-NodeJS/index.js]: make all requires at the top of file (preferably) or if that's not possible with functions, at the top of the function

File: Templates/FaceLocator-NodeJS/index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        .then(function (response) {
+            response = JSON.parse(response);
+
+            if (typeof response == 'undefined' && typeof response.faces.length === 0) {
+                context.done();
+                return;
+            }
+            context.log("From Vision Api:", response);
+
+            var guid = require('node-uuid');
```
Comment (mamaso) [inline Templates/FaceLocator-NodeJS/index.js]: same as above

File: Templates/FaceLocator-NodeJS/index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                    left: faceRectangle.left,
+                    top: faceRectangle.top,
+                    width: faceRectangle.width,
+                    height: faceRectangle.height
+                };
+                faces.push(face);
+            }
+            
+            context.bindings.outTable = faces;
+        }).catch(function (error) {
```
Comment (mamaso) [inline Templates/FaceLocator-NodeJS/index.js]: I pref the new arrow function syntax

```js
.catch((err) => context.log(error))
.finally(() => context.done())
```

**Comment by mamaso** [comment]
looks good in general, I just like to be nitty and concise :D

**Comment by soninaren** [comment]
All really good comments, Feel like a noob for node

File: Templates/FaceLocator-NodeJS/function.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
+﻿{
+    "bindings": [
+        {
+            "type": "blobTrigger",
+            "name": "image",
+            "path": "images/{name}.jpg",
+            "dataType": "binary",
```
Comment (mathewc) [inline Templates/FaceLocator-NodeJS/function.json]: Sweet - using the new DataType feature :)

**Pull request merged**